### ISSUE TITLE
How to generate a valid registration code?

### ISSUE THREAD

**Issue by sijanec** [ss23/entrust-identityguard-tools#1]
How can I emulate the Entrust app to generate the registration code in format ddddd-ddddd?

I tried a random one but my bank won't accept that. Is this known?

Thanks.

**Comment by JHthe4** [comment]
For anyone still asking themselves: Just use the actual Entrust app, and then give the code back to generate-otp.py

**Comment by sijanec** [comment]
Thank you for the reply, I realized that may be the only option. I guess I still need the actual app because it also sends some requests to some server, confirming that it had registered itself.

The app in question is not the official Entrust app, but it's a bank app that uses Entrust's system as a library.

I guess you'd need some translation service (website is written in Slovenian), if you wanted to read how I implemented this cracking algorithm in javascript: http://sijanec.eu/nlb.shtml

But you still need the app, that's something I haven't figured out yet how to bypass, is the HTTP "registration" also a known procedute or at least similar for all Entrust apps, or is it perhaps specific to implementors (my bank in this case)?

**Comment by JHthe4** [comment]
Hey, I think I might have misunderstood your question a bit. For me it was fine to install the app once for five minutes to generate my first code, and then just uninstall it again. Unfortunately, I'm not aware of a way to bypass it completely. Just wanted to say thank you for the great write-up though, I hope you manage to figure it out ;).

**Comment by sijanec** [comment]
Yeah, that's exactly what I did as well, though perhaps I didn't clarify this enough. Since then I didn't look at the app much do reverse engineer the initial procedure, as I didn't need it for bank login.

Thanks for the reply.
